---
title: "UMAP Clustering of Ribodepleted RNA-seq"
author: "Jenny Smith"
date: "4/14/20"
output: html_document
---


# Set-up

```{r setup, cache = FALSE, include = FALSE}
require(knitr)
knitr::opts_knit$set(root.dir = rprojroot::find_rstudio_root_file())
```

```{r}
knitr::opts_chunk$set(tidy.opts=list(width.cutoff=50),
                      tidy=TRUE,
                      fig.align='center',
                      fig.width = 10, fig.height = 10)


options(stringsAsFactors = FALSE,bitmapType = 'cairo')
grDevices::X11.options(type='cairo')
table = function (..., useNA = 'ifany') base::table(..., useNA = useNA)
```

```{r message=FALSE}
library(ggplot2)
library(ggrepel)
library(RColorBrewer)
library(tools)
library(gridExtra)
library(dendextend)

library(readr)
library(dplyr)
library(magrittr)
library(tibble)
library(tidyr)
library(data.table)
library(stringr)
library(DelayedArray)


library(DeGSEA)
getwd()
```

```{r}
# Define File/Data Locations
host <- Sys.info()[["nodename"]]
if(grepl("gizmo", host)){
  FASTDRIVE <- "/fh/fast/meshinchi_s"
}else{
  FASTDRIVE <- "/Users/jsmi26/fast_drive" #/fh/fast/meshinchi_s on the Fred Hutch HPC
}
RPROJ <- list(PROJHOME0=normalizePath(file.path(FASTDRIVE,"workingDir/TARGET/AML_TARGET/RNA/mRNAseq/analysis/")),
              CDE=normalizePath(file.path(FASTDRIVE, "workingDir/TARGET/AML_TARGET/Clinical/CDE/")),
              GENREFS=normalizePath(file.path(FASTDRIVE, "workingDir/TARGET/Reference_Data/")),
              TARGET=normalizePath(file.path(FASTDRIVE, "workingDir/TARGET/AML_TARGET/")),
              REFS=normalizePath(file.path(FASTDRIVE, "workingDir/0_For_Soheil/jlsmith3/reference_mapping-files/")),
              HOME=normalizePath(file.path(FASTDRIVE,"workingDir/0_For_Soheil/jlsmith3/RNA_seq_Analysis/")),
              SCRIPTS=normalizePath(file.path(FASTDRIVE,"workingDir/0_For_Soheil/jlsmith3/github_repos/")),
              WORKINGDIR=normalizePath(file.path(FASTDRIVE,"workingDir/")))
attach(RPROJ)
rm(RPROJ)
PROJHOME <- file.path(PROJHOME0,"2020.03.26_NUP98.Rearranged_Collab")
```


# Define Functions

```{r}
make_dends <- function(TMMCPM,method="ward.D2"){
  #TMM CPM must be log2 and already subset for the genes of interest
  d1 <- dist(t(TMMCPM), method = "euclidean", diag = FALSE,
             upper = FALSE) #sample distances WITHOUT SCALING
  d2 <- dist(TMMCPM, method = "euclidean", diag = FALSE,
             upper = FALSE) #gene distances WITHOUT SCaling

  samp.c1 <- hclust(d1, method = method, members = NULL) #sample clustering
  gene.c2 <- hclust(d2, method = method, members = NULL) #gene clustering


  list <- list(TMMCPM,samp.c1,gene.c2)
  names(list) <- c("TMMCPM","samp.c1", "gene.c2")

  return(list)
}
```

```{r}
make_alpha <- function(color, percent=100){
  # https://www.dataanalytics.org.uk/make-transparent-colors-in-r/
  ## Get RGB values for named color
  rgb.val <- col2rgb(color)
  
  ## Make new color using input color as base and alpha set by transparency
  t.col <- rgb(rgb.val[1], rgb.val[2], rgb.val[3],
               max = 255,
               alpha = (100 - percent) * 255 / 100)

}
```

# Read in the Clinical Data

```{r}
NUP98.cohort <- read.csv(file.path(PROJHOME,"TARGET_AML_NUP98.rearranged_Cleaned_Groups_REG_10.26.2020.csv")) %>% 
  mutate(Reg.=as.character(Reg.)) %>% 
  select(-USI)


dim(NUP98.cohort) #2304    4
# head(NUP98.cohort)
```

```{r}
chr13 <- read.csv(file.path(CDE, "Merged/00_Old/TARGET_AML_0531_1031_merged_CDEs_05.01.21.csv"),
                    na.strings = c("N/A","#N/A","NA","^$", "^\\.$")) %>% 
  select(matches("Reg.|chr13|_13"))


dim(chr13)
# chr13
```

```{r}
merged <- read.csv(file.path(CDE, "Merged/TARGET_AML_0531_1031_merged_CDEs_05.21.21.csv"), 
                    na.strings = c("N/A","#N/A","NA","^$", "^\\.$")) %>% 
  left_join(., chr13, by="Reg.")

inelig <- merged %>% 
  filter(Eligibility_Comments == "remove") %>% 
  pull(USI)

# head(merged)
dim(merged) 
```

```{r}
exon_juncs <- read.csv(file.path(PROJHOME,"NUP98_Fusions/Exon_Juncs/TARGET_AML_NUP98_fusion_exons_3.20.21.csv")) %>% 
  filter(!grepl("replicate", Sample))

dim(exon_juncs)
head(exon_juncs)
```

```{r}
CDEs <- merged %>% 
  filter(Eligibility_Comments != "remove") %>% 
  mutate(Reg.=as.character(Reg.)) %>%
  inner_join(., NUP98.cohort,
             by="Reg.") %>%  
  left_join(., select(exon_juncs,-c(NUP98.Rearranged.Groups:NUP98.Rearranged), -Reg., -FIX),
            by="USI") %>% 
  mutate(CNS.Disease.Harmonized=case_when(
    grepl("CNS[12]", CNS.disease.at.on.study) ~ "No",
    grepl("CNS3", CNS.disease.at.on.study) ~ "Yes",
    grepl("Yes|No", CNS.disease) ~ CNS.disease,
    TRUE ~ "Unknown")) %>% 
  
  #For this analysis include <0.1 AR are FLT3-ITD positive 
  mutate_at(vars(FLT3.ITD.positive.), ~gsub("<0.1", "Yes", .)) %>% 
  mutate(Any_chr13_Abnormality=case_when(
                  !is.na(deletion_chr13)  | !is.na(translocation_13) ~ "chr13_Abnormality", 
                  grepl("monosomy13", monosomy_trisomy_13) ~ "chr13_Abnormality",
                  ISCN=="Unknown" ~ "Unknown",
                  TRUE ~ "None"), 
         Any_chr13_Deletion=case_when(
           !is.na(deletion_chr13) ~ "chr13_deletion",
           ISCN=="Unknown" ~ "Unknown",
           TRUE ~ "None"),
         Any_chr13_CNV=case_when(
           !is.na(deletion_chr13) | !is.na(monosomy_trisomy_13) ~ "chr13_CNV",
           ISCN=="Unknown" ~ "Unknown",
           TRUE ~ "None")) %>% 
  mutate(deletion_chr13_groups=case_when(
              ISCN == "Unknown" ~ "Unknown",
              is.na(deletion_chr13) ~ "OtherAML",
              TRUE ~ "del(13)")) %>% 
  mutate(monosomy_trisomy_13_groups=case_when(
              ISCN == "Unknown" ~ "Unknown",
              is.na(monosomy_trisomy_13) ~ "OtherAML",
              TRUE ~ as.character(monosomy_trisomy_13))) %>% 
  mutate(translocation_13_groups=case_when(
              ISCN == "Unknown" ~ "Unknown",
              is.na(translocation_13) ~ "OtherAML",
              TRUE ~ "t(13;X)")) 

dim(CDEs) #2296  198
```



# Read in the counts data

```{r}
updated.IDs <- read.csv(file.path(PROJHOME0,"0000.00.02_Reference_GeneInfo/GeneSymbol_Ensembl_ID_Conversion_GRCh37_v69_to_v102.csv")) %>% 
  rename_all(~c("gene_id","geneSymbol_Updated"))

dim(updated.IDs)
head(updated.IDs)
```

```{r}
ID.map <- read.csv(file.path(PROJHOME0,"0000.00.02_Reference_GeneInfo/GeneSymbol_Ensembl_ID_Conversion_GRCh37.69_FromBCCA.csv")) %>% 
  left_join(.,updated.IDs, by= "gene_id")

dim(ID.map) #58450     3
head(ID.map)

# filter(ID.map, geneSymbol != geneSymbol_Updated) %>%  dim() #6,235 genes have updated symbols. 

# filter(ID.map, is.na(geneSymbol_Updated)) %>% dim() #1,440 genes dont have mapping to its ensembl ID.
#   write.csv(., "Missing_Ensembl_IDs.csv", row.names = F)
```

```{r}
# cts <- readRDS(file.path("/fh/fast/meshinchi_s/workingDir/TARGET/AML_TARGET/RNA/mRNAseq/analysis/2019.12.31_UMAP_Clustering/Expression_Data/TARGET_AML_DSAML_MPN_NBM_Ribodepleted_dupGenesRemoved_Fractionalcounts.RDS"))

cts <- readRDS(file.path(PROJHOME0, "0000.00.03_ExpressionMatrices/BCCA_GRCh37_Ensembl_v69/TARGET_AML_MPN_DS_NBM_3044Samples_Ribodepleted_RNAseq_geneLevel_dupGenesRemoved_FractionalCounts.RDS"))

cts <- as.data.frame(cts)
gene_ids <- cts[,1:2]
rownames(cts) <- cts$geneSymbol
cts <- cts[,-c(1:2)]



keep <- rowSums(cts)>= 10
cts <- cts[keep, ]

dim(cts) #51573  1575
head(cts[,1:5])
```

```{r eval=FALSE}
TPMs <- readRDS(file.path(PROJHOME0, "0000.00.03_ExpressionMatrices/BCCA_GRCh37_Ensembl_v69/TARGET_AML_MPN_DS_NBM_3044Samples_Ribodepleted_RNAseq_geneLevel_dupGenesRemoved_TPM.RDS"))


TPM <- as.data.frame(TPMs)
gene_ids <- TPM[,c(1:2)]
rownames(TPM) <- TPM$geneSymbol
TPM <- TPM[,-c(1:2)]



head(TPMs[,1:5])
dim(TPMs) # 51573  
```


# Define Samples 

```{r}
sample_info <- read.csv(file.path(TARGET, "SequencingDataMatrix/00_archive/TARGET_AML_Ribodepleted_Manifest_10.08.20.csv")) %>% 
  filter(!USI %in% inelig) %>% 
  filter(USI %in% CDEs$USI |  grepl("CD34_PB|NBM", Group)) %>% 
  filter(!grepl("replicate", Sample, ignore.case = T),
         !grepl("TARGET.20.PAXLWH\\.[A-Z]",Sample),
         !grepl("FlowSorted",Group)) %>% 
  
  left_join(., dplyr::select(CDEs,USI,Reg., 
                             NUP98.Rearranged.Groups, NUP98.Rearranged,
                             Age.Category, matches("Age|^OS|^EFS|event|time|mutation"), 
                             SNVs,
                            FLT3.ITD.positive.,Treatment.Arm,
                            Mutations.Category,M7_AML,M6_AML,
                            ISCN,matches("chr13|_13")),
            by=c("USI")) %>%
  
  #Chr13 Groups
  # left_join(., select(chr13, -c(USI:NUP98.Rearranged)), 
  #           by="Reg.") %>% 
  select(-Reg.) %>%
  mutate_at(vars(matches("Any_chr13")), ~case_when(
    Group == "AML"  & ISCN == "Unknown" ~ "Unknown",
    Group == "AML" & is.na(.) ~ "Unknown",
    Group != "AML" & is.na(.)  ~ Group,
    TRUE ~ .)) %>% 
  
  
  mutate_at(vars(Mutations.Category, Age.Category,M7_AML,
                 NUP98.Rearranged.Groups,
                 matches("(deletion|monosomy|[Tt]ranslocation|Any_chr13).+groups$")),
            ~ifelse(is.na(.), Group, .)) %>%
  mutate_at(vars(AML_Subtype),
            ~ifelse(NUP98.Rearranged.Groups=="NUP98-X", "NUP98-X", .)) %>%
  mutate(Age.Category=factor(Age.Category, levels=c("Less than 3 years",
                                                    "Between 3 and 5 years",
                                                    "Between 5 and 10 years",
                                                    "Between 10 and 18 years",
                                                    "Greater than 18 years",
                                                    "Unknown",
                                                    "NBM",
                                                    "CD34_PB"))) %>%
  set_rownames(.$Sample)

dim(sample_info) #2238  
# head(sample_info)
table(sample_info$AML_Subtype, useNA = 'ifany')
table(sample_info$Any_chr13_Deletion, useNA = 'ifany')
table(sample_info$Time_point)
```



## Define Contrasts 

```{r}
samps_dx_rlps <- sample_info %>% 
  filter(grepl("NUP98", NUP98.Rearranged))

table(samps_dx_rlps$NUP98.Rearranged.Groups, 
      samps_dx_rlps$Time_point, useNA='ifany')
```

```{r}
samps_nup_majorFusions <- sample_info %>%
  filter(grepl("NBM|CD34_PB|diagnostic",Time_point)) %>% 
  filter(!grepl("AML|No.Primary.Fusion", AML_Subtype))

table(samps_nup_majorFusions$AML_Subtype)
table(samps_nup_majorFusions$Any_chr13_Deletion, useNA = "ifany")
```

```{r}
#removing TARGET.20.PAXLWH.CD34NEG.01R and other associated experimenal samples
samps_NUP98 <- filter(sample_info,
                grepl("NUP98", NUP98.Rearranged.Groups),
                grepl("diagnostic", Time_point, ignore.case = T))

table(samps_NUP98$NUP98.Rearranged.Groups, useNA='ifany')
# NUP98-KDM5A  NUP98-NSD1     NUP98-X 
#          32         104          20

table(samps_NUP98$AML_Subtype, samps_NUP98$NUP98.Rearranged.Groups)
length(unique(samps_NUP98$Sample))
# dim(samps_NUP98) #156  42
# table(samps_NUP98$Mutations.Category,samps_NUP98$AML_Subtype)
# table(samps_NUP98$Age.Category,samps_NUP98$AML_Subtype)
# table(samps_NUP98$M6_AML,samps_NUP98$AML_Subtype)
```

```{r}
Cols <- c("NUP98.Rearranged.Groups", "Group",
          "AML_Subtype","Mutations.Category",
          "Batch","Time_point","Tissue",
          "Age.Category","M7_AML",
          "Protocol", 
          grep("(deletion|monosomy|[Tt]ranslocation).+groups$|^Any_", colnames(sample_info), value=T))
# Cols
```


#Colors for Plotting

```{r fig.height=2, fig.width=15}
colors37 = c("#466791","#60bf37","#953ada","#4fbe6c","#ce49d3","#a7b43d","#5a51dc","#d49f36","#552095","#507f2d","#db37aa","#84b67c","#a06fda","#df462a","#5b83db","#c76c2d","#4f49a3","#82702d","#dd6bbb","#334c22","#d83979","#55baad","#dc4555","#62aad3","#8c3025","#417d61","#862977","#bba672","#403367","#da8a6d","#a79cd4","#71482c","#c689d0","#6b2940","#d593a7","#895c8b","#bd5975")

barplot(rep(1,37), col=colors37, names.arg = colors37, las=2)
```

```{r fig.height=4}
subtype.levels <- c("AML", 
                    "CBFA2T3-GLIS2", 
                    "CBFB-MYH11",
                    "ETS-Fusion",
                    "NBM",
                    "DEK-NUP214", 
                    "del5q", 
                    "ERG-HNRNPH1",
                    "ETV6-MNX1", 
                    "FUS-ERG",
                    "KAT6A-CREBBP",
                    "KMT2A",
                    "monosomy7",
                    "No.Primary.Fusion",
                    "NUP98-KDM5A",
                    "NUP98-NSD1",
                    "NUP98-X",
                    "RBM15-MKL1",
                    "RUNX1-CBFA2T3",
                    "RUNX1-RUNX1T1",
                    "CD34_PB")

df <- sample_info %>% 
  dplyr::select(all_of(Cols)) %>%
  mutate(AML_Subtype=factor(AML_Subtype, levels=subtype.levels))

cc <- colorCodes_aheatmap(df=df)
cc <- lapply(cc, function(x){x[["AML"]] <- "grey80"; return(x)})
cc <- lapply(cc, function(x){x[["NBM"]] <- "white"; return(x)})
cc <- lapply(cc, function(x){x[["CD34_PB"]] <- "ivory2"; return(x)})
cc$M7_AML["No"] <- "mistyrose3"
cc$M7_AML["Yes"] <- "red1"
cc$M7_AML["Unknown"] <- "lightblue"



#Manually Update AML subtype colors
cc[c(1,3)] <- lapply(cc[c(1,3)], function(x){
  x[["NUP98-X"]] <- "green1"
  x[["NUP98-KDM5A"]] <- "magenta"
  x[["NUP98-NSD1"]] <- "steelblue1"
  return(x)})

cc$AML_Subtype[["No.Primary.Fusion.CNV"]] <- "azure4"
cc$AML_Subtype[["RUNX1-RUNX1T1"]] <- "sienna4"
cc$AML_Subtype[["CBFB-MYH11"]] <- "red"
cc$AML_Subtype[["KMT2A"]] <- "khaki2"
cc$Mutations.Category[["WT1.mutation."]] <- "yellow"



par(mar=c(14,4,4,2))
barplot(rep(1,length(cc$AML_Subtype)), col=cc$AML_Subtype, 
        names.arg=names(cc$AML_Subtype),las=2)
# barplot(rep(1,length(cc$Mutations.Category)), col=cc$Mutations.Category,
#         names.arg=names(cc$Mutations.Category),las=2)

# barplot(rep(1,length(cc$NUP98.Rearranged.Groups)),
#         col=cc$NUP98.Rearranged.Groups,
#         names.arg=names(cc$NUP98.Rearranged.Groups),
#         las=2)

# saveRDS(cc,"UMAP/NUP98_Only/UMAP_NUP98_ColorCodes_8.01.20.RDS")
```

```{r}
# saveRDS(cc,"UMAP_NUP98_ColorCodes_7.16.20.RDS")
```


# UMAP NUP98 Diagnostic Only

```{r}
cts_NUP98 <- cts[, samps_NUP98$Sample]
cts_NUP98 <- as.matrix(cts_NUP98[rowSums(cts_NUP98)>= 10, ])

dim(cts_NUP98) #38247   156
```

```{r}
#TFIDF TRANSFORMED Counts
# Term Frequency - Inverse Document Frequency (TF-IDF) 
# https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6101073/
cell_total <- apply(cts_NUP98, 2, sum)
geomean <- exp(mean(log(cell_total)))
sf <- cell_total/geomean
sf.scaled <- t(t(cts_NUP98)/sf)
tf_NUP98 <- seqGlue::tf_idf_transform(log2(sf.scaled+1))
# Mean vs Dispersion Feature Selection 
obj <- seqGlue::calc_dispersion(cts_NUP98, removeOutliers = TRUE) #removes outlier genes/transcripts based on cooks distance
sg_NUP98 <- seqGlue::get_selected_genes(seqGlue::select_genes(obj, top_n=NULL))
tf_NUP98 <- tf_NUP98[sg_NUP98,]

dim(tf_NUP98) #10353   156
```

```{r eval=FALSE}
sample_info_NUP98 <- sample_info[samps_NUP98$Sample,]
uidl <- paste(c(dim(tf_NUP98),"NUP98","log2"),collapse = "_")

umap_NUP98 <- UMAP_workflow(TFIDF_Matrix = tf_NUP98log, 
                          samples_vector = c$Sample,
                          sample_info_df = sample_info_NUP98,
                          min_dist = 0.0001,
                          n_neighbors=2,
                          cc = cc, 
                          k2=10,
                          res2=0.025,
                          uniqID=uidl)

length(umap_NUP98$input_features)
# saveRDS(umap_NUP98,paste0("TARGET_AML_sg",length(umap_NUP98$input_features),"_NUP98.Rearr_log2Counts_UMAP_NUP98_Results_7.16.20.RDS"))
```

```{r}
sample_info_NUP98 <- sample_info[samps_NUP98$Sample,]

umap_NUP98 <- readRDS(file.path(PROJHOME,"UMAP/NUP98_Only/TARGET_AML_sg8780_NUP98.Rearr_log2Counts_UMAP_NUP98_Results_7.16.20.RDS"))

#update clinical covars
umap_NUP98$umap_res <- umap_NUP98$umap_res %>% 
  select(USI,x:z,matches("^cluster")) %>% 
  left_join(.,select(sample_info_NUP98, everything()),
            by="USI") %>% 
    filter(!USI %in% inelig) %>% 
  
  mutate(chr13_Abn_groups=case_when(
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "NUP98-X" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "NUP98-KDM5A" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "OtherAML" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    TRUE ~ NUP98.Rearranged.Groups)) %>% 
  set_rownames(.$Sample)

# length(umap_NUP98$input_features)
# sapply(umap_NUP98$umap_res %>% select(matches("^cluster")), table, useNA='ifany')
# write.csv(umap_NUP98$umap_res, "UMAP/NUP98_Only/TARGET_AML_NUP98.R_only_sg8780.csv",row.names = F)


# umap_NUP98$umap_model$n_neighbors
# umap_NUP98$umap_model$method
# umap_NUP98$umap_model$search_k
```

```{r}
# table(res$cluster_k5, res$NUP98.Rearranged.Groups)
```

```{r}
NUP98_colors <- cc$NUP98.Rearranged[unique(umap_NUP98$umap_res$NUP98.Rearranged.Groups)]
par(mar=c(8,4,4,2))
barplot(rep(1,length(NUP98_colors)), col=NUP98_colors, 
        names.arg = names(NUP98_colors), las=2)
```

### 2D Scatters 

```{r}
table(umap_NUP98$umap_res$Any_chr13_Abnormality, 
      umap_NUP98$umap_res$NUP98.Rearranged.Groups)

table(umap_NUP98$umap_res$chr13_Abn_groups)

colors_del13 <- c("NUP98-KDM5A" = "magenta",
                  "NUP98-KDM5A/chr13_abn" = "darkorchid4",
                 "NUP98-NSD1" = "steelblue1",
                 "NUP98-NSD1/chr13_abn"="steelblue4",
                 "NUP98-X" = "green1",
                 "NUP98-X/chr13_abn"="darkgreen",
                 "OtherAML" = "grey80",
                 "OtherAML/chr13_abn"="cyan3")
```

```{r}
labs <- umap_NUP98$umap_res %>% 
  group_by(chr13_Abn_groups) %>% 
  dplyr::count() %>% 
  mutate(labs=paste(chr13_Abn_groups, paste0("(N=",n,")"))) %>% 
  pull(labs)

# labs
```

```{r fig.height=8, fig.width=7}
nup_chr13 <- ggplot(umap_NUP98$umap_res, aes(x=x, y=y, fill=chr13_Abn_groups)) +
  geom_point(shape=21,size=3.5, alpha=0.65, color="grey70") +
  scale_color_manual(values=colors_del13) +  # labels=labs
  scale_fill_manual(values=colors_del13) + # labels=labs
  labs(x="UMAP 1", y="UMAP 2", 
       color="NUP98 Fusions",
       fill="NUP98 Fusions"
       # title="NUP98 Fusions with Chr13 Abnormalities"
       ) +
  guides(color = guide_legend(override.aes = list(size=6), ncol=2, 
                                title.position="top", title.hjust = 0.5),
         fill = guide_legend(override.aes = list(size=6), ncol=2, 
                             title.position="top", title.hjust = 0.5)) +

  theme_classic() +
  theme(axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18), 
        legend.text = element_text(size=14), 
        legend.title = element_text(size=14), 
        plot.margin = margin(r=4,t=4,l=4, unit="mm"),
        legend.position = "bottom")

nup_chr13
# ggsave(plot = nup_chr13, filename = "Figures/Chr13_deletions/TARGET_AML_NUP98_Fusions_UMAP_chr13_Abnormalities.pdf", height = 7, width = 6, device = "pdf")
```

```{r fig.height=12, fig.width=20}
# pdf("UMAP/NUP98_Only/TARGET_AML_NUP98.Rearranged_2D_Scatterplots.pdf", height = 15, width = 22)
grid.arrange(grobs=list(umap_NUP98$umap_2D_scatter$NUP98.Rearranged.Groups,
                       umap_NUP98$umap_2D_scatter$Mutations.Category,
                       umap_NUP98$umap_2D_scatter$Age.Category,
                       umap_NUP98$cluster_plots2$scatter,
                       umap_NUP98$cluster_plots2$barplot), ncol=3)
# dev.off()
```

```{r fig.width=10.5, fig.height=10}
# pdf("UMAP/NUP98_Only/TARGET_AML_NUP98.Rearranged_2D_Scatterplots_forASH.pdf", height = 10, width = 10.5)
grid.arrange(grobs=list(umap_NUP98$umap_2D_scatter$NUP98.Rearranged.Groups,
                       umap_NUP98$umap_2D_scatter$Mutations.Category,
                       umap_NUP98$umap_2D_scatter$Age.Category,
                       umap_NUP98$umap_2D_scatter$M7_AML), 
             ncol=2, nrow=2)

# dev.off()
```

```{r fig.height=7, fig.width=4.5}
# pdf("UMAP/NUP98_Only/TARGET_AML_NUP98.Rearranged_LeidenCluster_Barplot_forASH_flipped.pdf",
    # height = 7, width = 4.5)
umap_NUP98$cluster_plots2$barplot +
  labs(y="Number of Samples") +
  theme(plot.margin = margin(l=2, r=0, unit="cm"),
        legend.direction = 'vertical',
        legend.justification = 'right',
        legend.position = 'top',
        legend.text = element_text(size=14), 
        legend.title=element_text(size=14, color="black"), 
        axis.title = element_text(size=18, color="black"),
        axis.text = element_text(size=16, color="black"))+
  coord_flip()
# dev.off()
```

### 3D Plot.ly 

```{r message=FALSE}
library(plotly)
```

```{r}
res <- umap_NUP98$umap_res 

info <- paste(res$Sample, 
              paste0("Mutation Category: ",res$Mutations.Category),
              sep="\n") %>% 
  ifelse(!is.na(res$Primary.Fusion.CNV), 
         paste(., paste0("Primary Fusion CNV: ",res$Primary.Fusion.CNV), sep="\n"), .) %>% 
  ifelse(!is.na(res$Overlap.Mutation.Info),
         paste(., gsub("/", "\n", res$Overlap.Mutation.Info), sep="\n"), .) %>%
  gsub("OtherAML","", .) %>%
  ifelse(!is.na(res$Age.Category), 
         paste(., paste0("Age Category: ", res$Age.Category), sep="\n"), .) %>%
  ifelse(!is.na(res$EFS.event.type.ID),
         paste(., paste0("Event: ", res$EFS.event.type.ID), sep="\n"), .)  %>%
  gsub("Unknown","",.) %>%
  set_names(., res$Sample)


length(info)
# info
```

```{r}
# cc.clusters <- colorCodes_aheatmap(df=select(res,matches("^cluster")),T)
# cc.clusters
cc.clusters <- set_names(c("red","dodgerblue","forestgreen","darkorchid3", "gold"),
                            levels(res$cluster_k5))
```

```{r}
expand_coords <- function(x,y){ifelse(x>0, x+y, x-y)}
ranges <- lapply(res[,c("x","y","z")], range)

# ranges[[1]] <- c(ranges$x[1], ranges$x[2])
# ranges[2:3] <- lapply(ranges[2:3], expand_coords,y=0)
# ranges

m <- list(
  l = 50,
  r = 50,
  b = 50,
  t = 50,
  pad = 200
)
```

```{r fig.width=16}
textcol <- rgb(1,1,1)
bgcol <- rgb(0,0,0)

p.NUP98 <- plot_ly() %>% 
  #plot diagnostic and normal samples
  add_trace(data=res,
            x = ~x, y = ~y, z = ~z,
            # color =  ~NUP98.Rearranged.Groups, 
            color = ~cluster_k5,
            # colors = cc$NUP98.Rearranged.Groups,
            colors = cc.clusters,
            type='scatter3d',
            mode='markers',
            text=info,
            hoverinfo='text',
            showlegend=TRUE,
            marker=list(size=4.5,
                        opacity=0.7),
                        # line=list(color=col2hex("grey80"),
                        #           width=0.25)),
            # line=list(size=1, color=col2rgb("grey80")),
            inherit = TRUE) %>%
  # title=list(text="Pediatric NUP98-Rearranged AML Clustering By Gene Expression",
  layout(font = list(color=textcol,
                                size=10),
         margin = m,
         scene = list(xaxis = list(title = 'UMAP_1',
                                   color=textcol,
                                   range=ranges$x,
                                   size=10,
                                   backgroundcolor=bgcol,
                                   showbackground=TRUE,
                                   showgrid=TRUE,
                                   gridcolor=textcol,
                                   tickcolor=textcol),
                     yaxis = list(title = 'UMAP_2',
                                  color=textcol,
                                  range=ranges$y,
                                  size=18,
                                  backgroundcolor=bgcol,
                                  showbackground=TRUE,
                                  showgrid=TRUE,
                                  gridcolor=textcol,
                                  tickcolor=textcol),
                     zaxis = list(title = 'UMAP_3',
                                  color=textcol,
                                  range=ranges$z,
                                  size=10,
                                  backgroundcolor=bgcol,
                                  showbackground=TRUE,
                                  showgrid=TRUE,
                                  gridcolor=textcol,
                                  tickcolor=textcol),
                     bgcolor=bgcol,
         legend=list(font=list(size=12, color=textcol),
                     itemsizing="constant"),
         plot_bgcolor=bgcol,
         paper_bgcolor=bgcol))
 
p.NUP98
```

```{r}
htmlwidgets::saveWidget(as_widget(p.NUP98), "~/0000_ASH_Abstracts/2020/TARGET_NUP98.R_byGroup_UMAP.html",background = bgcol)
```

```{r}
table(res$cluster_k5, res$NUP98.Rearranged.Groups)

res %>% 
  group_by(cluster_k5,NUP98.Rearranged.Groups) %>%
  summarize(N=n(),
            Med=median(Age.in.years, na.rm = T), 
            Min=min(Age.in.years, na.rm = T), 
            Max=max(Age.in.years, na.rm = T))
  
  
  
```

```{r}
# install.packages("orca")
# install.packages("processx")
# library(orca)
# library(processx)
# orca(p.NUP98,"NUP98_alone_3dscatter_A.pdf", format = "pdf")
```



### 3D Static
 
```{r}
# devtools::install_github("AckerDWM/gg3D")
library(gg3D)
library(cowplot)
# library(svglite)
```

```{r fig.height=7, fig.width=10}
theta=100
phi=-10


bgCol <- "white"
txtCol <- "black"

p <- ggplot(data=select(res,x,y,z, NUP98.Rearranged.Groups, 
                        cluster_k5),
            aes(x=x, y=y,z=z, fill=cluster_k5, color=NUP98.Rearranged.Groups)) +
  theme_void() +
  axes_3D(phi=phi,theta=theta) +
  stat_3D(phi=phi,theta=theta,size=5, alpha=0.75,
          shape=19, stroke=1.5)

# p
```


```{r}
output="png"
ggsave(filename = paste0("",
                         output),
       plot=rotated,
       height = 7, width=10,
       units="in",
       dpi=300,
       device=output)

```


### Marker Genes per Cluster

```{r}
# dir(file.path(SCRIPTS,"RNAseq_Analysis/Analysis/UMAP_Clustering/umap_workflow_find_markers.R"))
source(file.path(SCRIPTS,"RNAseq_Analysis/Analysis/UMAP_Clustering/UMAP_workflow_find_markers.R"))
```

```{r}
res <- umap_NUP98$umap_res 

#Groupings/clusterings for marker gene identification per cluster
cell_group_df <- select(res, Sample,cluster_k5) %>% 
  set_colnames(c("cell_id","cell_group")) %>%
  arrange(desc(cell_group)) %>%
  as.data.frame()

rownames(cell_group_df) <- cell_group_df$cell_id

head(cell_group_df)
# dim(cell_group_df) #156   2
```

```{r}
dim(sf.scaled)
range(sf.scaled) #0 37,932,406
range(cts_NUP98) #0 53,941,548
```

```{r}
#binary  "normalized" counts. Using 0.5 as the cut-off rather than 0. 
bin.mat <- cts_NUP98 > 0.5
# head(bin.mat[,1:5])
class(bin.mat)
```

```{r}
#Aggregate by mean the lib size scaled counts
agg_mat <- sf.scaled[,cell_group_df[,1]]
agg_mat <- my.aggregate.Matrix(x=Matrix::t(agg_mat), groupings = as.factor(cell_group_df[,2]), fun="mean")
agg_mat <- as.matrix(Matrix::t(agg_mat))

dim(agg_mat) #38294     5
head(agg_mat[,1:5])
```

```{r}
#Aggregate binary matrix per sample cluster
bin_mat = bin.mat[,cell_group_df[,1]]
bin_mat = my.aggregate.Matrix(Matrix::t(bin_mat),
                              as.factor(cell_group_df[,2]),
                              fun="mean")
bin_mat = Matrix::t(bin_mat)
head(bin_mat[,1:5])
dim(bin_mat) #38294     5
```

```{r}
# Now compute a Jensen Shannon specificity score for each gene w.r.t each group
cluster_spec_mat <- specificity_matrix(agg_expr_matrix = agg_mat, cores=1)
cluster_marker_score_mat <- as.matrix(bin_mat * cluster_spec_mat)


# head(cluster_spec_mat[,1:5])
head(cluster_marker_score_mat[,1:5])
dim(cluster_spec_mat) #38294     5
dim(cluster_marker_score_mat) #38294     5
```

```{r}
#("Gathering score tables")
cluster_marker_score_table = tibble::rownames_to_column(as.data.frame(cluster_marker_score_mat))
cluster_marker_score_table = tidyr::gather(cluster_marker_score_table,
                                           "cell_group", "marker_score", -rowname)

cluster_spec_table = tibble::rownames_to_column(as.data.frame(cluster_spec_mat))
cluster_spec_table = tidyr::gather(cluster_spec_table, "cell_group", "specificity", -rowname)

cluster_expr_table = tibble::rownames_to_column(as.data.frame(as.matrix(agg_mat)))
cluster_expr_table = tidyr::gather(cluster_expr_table, "cell_group", "mean_expression", -rowname)

cluster_fraction_expressing_table = tibble::rownames_to_column(as.data.frame(as.matrix(bin_mat)))
cluster_fraction_expressing_table = tidyr::gather(cluster_fraction_expressing_table, "cell_group", "fraction_expressing", -rowname)
```


```{r}
cluster_marker_score_table$specificity = cluster_spec_table$specificity
cluster_marker_score_table$mean_expression = cluster_expr_table$mean_expression
cluster_marker_score_table$fraction_expressing = cluster_fraction_expressing_table$fraction_expressing

head(cluster_marker_score_table)
table(cluster_marker_score_table$cell_group)
dim(cluster_marker_score_table) #191470      6
```

```{r}
#The data frame marker_test_res contains a number of metrics for how specifically expressed each gene is in each partition. We could group the cells according to cluster, partition, or any categorical variable in colData(cds). You can rank the table according to one or more of the specificity metrics and take the top gene for each cluster. For example, pseudo_R2 is one such measure.
genes_to_test_per_group=2000
cluster_marker_score_table_ntop <- cluster_marker_score_table %>%
  dplyr::group_by(cell_group) %>%
  dplyr::top_n(genes_to_test_per_group, marker_score)

dim(cluster_marker_score_table_ntop)
```

```{r}
cell_group_df$cell_id <- as.character(cell_group_df$cell_id)
cell_group_df$cell_group <- as.character(cell_group_df$cell_group)

cores=1
old_omp_num_threads=1
old_blas_num_threads=1
reference_cells=NULL


#speedglm.maxiter=25 option for test_marker_for_cell_group
marker_test_res = pbmcapply::pbmcmapply(FUN=test_marker_for_cell_group,
                                        cluster_marker_score_table_ntop$rowname,
                                        cluster_marker_score_table_ntop$cell_group,              
                                        MoreArgs=list(cell_group_df, 
                                                      sf.scaled, 
                                                      reference_cells), 
                                        ignore.interactive = TRUE, 
                                        mc.cores=cores)


# str(marker_test_res)
dim(cluster_marker_score_table_ntop)
dim(marker_test_res)
```

marker_test_res$warning
<simpleWarning in speedglm::speedglm(is_member ~ f_expression, acc = 0.001, model = FALSE,     y = FALSE, verbose = TRUE, family = stats::binomial()): Maximum number of iterations reached without convergence> 

got this warning with 25 and 50 iterations. So I manually increased the iterations to 100. That has resolved the warning for now. 

Monocle3 updated thier code on the repo to include maxit as a parameter, but I will just do it as 50 for now. 


```{r}
marker_test_res = t(marker_test_res)
marker_test_res = as.matrix(marker_test_res)
colnames(marker_test_res) = c("pseudo_R2", "lrtest_p_value")

# str(marker_test_res)
head(marker_test_res)
dim(marker_test_res)
```

```{r}
marker_test_res = dplyr::bind_cols(cluster_marker_score_table_ntop,
                                   as.data.frame(marker_test_res))

marker_test_res$lrtest_q_value =
  stats::p.adjust(marker_test_res$lrtest_p_value,
                  method="bonferroni",
                  n=length(cluster_spec_mat))

marker_test_res = marker_test_res %>% dplyr::select(rowname,
                                                    cell_group,
                                                    marker_score,
                                                    mean_expression,
                                                    fraction_expressing,
                                                    specificity,
                                                    pseudo_R2,
                                                    lrtest_p_value,
                                                    lrtest_q_value)
marker_test_res = marker_test_res %>%
  dplyr::rename(gene_id=rowname,marker_test_p_value=lrtest_p_value,
                marker_test_q_value=lrtest_q_value)

marker_test_res$pseudo_R2 = unlist(marker_test_res$pseudo_R2)
marker_test_res$marker_test_p_value=unlist(marker_test_res$marker_test_p_value)

dim(marker_test_res) #10000     9
# write.csv(marker_test_res,"UMAP/NUP98_Only/TARGET_AML_NUP98.R_UMAP_Cluster_Gene_Markers.csv", row.names = FALSE)
```


####Plot UMAP projection by top gene markers

```{r}
TPM <- readRDS(file.path(PROJHOME,"0000.00.03_ExpressionMatrices/TARGET_AML_DSAML_MPN_NBM_Ribodepleted_dupGenesRemoved_TPM.RDS"))

TPM_NUP98 <- TPM[,samps_NUP98$Sample]

dim(TPM_NUP98)
```

```{r}
marker_test_res <- read.csv(file.path(PROJHOME,"UMAP/NUP98_Only/TARGET_AML_NUP98.R_UMAP_Cluster_Gene_Markers.csv"))


head(marker_test_res)
dim(marker_test_res) #10,000     9
```

```{r}
unique_gmarkers <- tibble()
for (i in 1:5){
  gm <- filter(marker_test_res, cell_group==i) 
  others <- filter(marker_test_res, cell_group != i )
  
  uniq <- gm$gene_id[!gm$gene_id %in% others$gene_id]  
  
  unique_gmarkers <- marker_test_res %>% 
    filter(cell_group == i, gene_id %in% uniq) %>% 
    bind_rows(., unique_gmarkers)
  rm(gm, uniq, others)
    
}

dim(unique_gmarkers) #8433    9
table(unique_gmarkers$cell_group)
```

```{r}
options(scipen = 999)
forplot <- unique_gmarkers %>% 
  group_by(cell_group) %>% 
  filter(marker_test_q_value < 0.001) %>%
  arrange(desc(specificity)) %>%
  slice(1:10) %>% 
  ungroup() %>%
  select(gene_id, cell_group,marker_score,specificity, pseudo_R2,marker_test_q_value )

head(forplot)
View(forplot)
any(duplicated(forplot$gene_id))
```

```{r}
cts_to_plot <- TPM_NUP98[forplot$gene_id,]%>% 
  as.data.frame() %>%
  rownames_to_column("gene_id") %>%
  filter(gene_id %in% forplot$gene_id) %>%
  gather(Sample,TPM,-gene_id)  %>%
  
  group_by(gene_id) %>%
  filter(max(TPM) > 10) %>% #select more highly expressed genes to ensure biological relevance
  ungroup() %>%

  mutate(log2_TPM=log2(TPM+1)) %>%
  left_join(., forplot, by=c("gene_id")) %>%
  left_join(., select(res, Sample,x,y,z),
            by="Sample") %>%
  arrange(desc(cell_group))

head(cts_to_plot)
# dim(cts_to_plot)

# table(cts_to_plot$gene_id, cts_to_plot$cell_group)
```

```{r}
expn_plots <- lapply(forplot$gene_id, expression_scatter_2D, norm_counts=cts_to_plot)
names(expn_plots) <- forplot$gene_id
```

```{r}
for(i in 1:5){
  clust_gs <- filter(forplot, cell_group==i, 
                     gene_id %in% cts_to_plot$gene_id) %>% 
    pull(gene_id)
  
  dirname = paste0("Figures/NUP98_UMAP_Expression_Scatter2D_Cluster",i,".pdf")
  dir.create(dirname, recursive = T)

  lapply(clust_gs, 
         function(x) ggsave(filename = paste0(dirname,"/", x, "_Cluster",i,".pdf"), 
                            plot=expn_plots[[x]], device = "pdf", height = 5, width = 5))
  
  rm(clust_gs, dirname)
}
```



##GSEA with top Gene Markers 

```{r}
# BiocManager::install("pathfindR")
library(pathfindR)
```

```{r eval=FALSE}

results <- list()
input <- unique_gmarkers
# input <- marker_test_res
geneset="Reactome"
for(clust in 1:5){
 
  clust_genes <- input %>% 
    filter(cell_group==clust) %>% 
    ungroup() %>% 
    select(gene_id, specificity, marker_test_p_value) %>%
    as.data.frame()
  
  outdir=paste0("UMAP/NUP98_Only/","Cluster",clust,"_uniqGenes_",geneset)
  dir.create(outdir,recursive = TRUE)
  filename <- paste(outdir,
                    paste0("UMAP_NUP98.R_Cluster",clust,"_uniqGenes_", geneset,"_paths.csv"), 
                    sep="/")
  # print(filename)
  print(outdir)
  
  gsea <- run_pathfindR(input=clust_genes,
                               p_val_threshold=1,
                               gene_sets = geneset,
                               min_gset_size = 20,
                               max_gset_size = 400,
                               pin_name_path = "Biogrid",
                               visualize_enriched_terms = FALSE,
                               max_to_plot = 0,
                               plot_enrichment_chart = FALSE,
                               output_dir = outdir,
                               search_method="GR")

  results[[paste0("Cluster_", clust)]] <- gsea
  
  write.csv(gsea,filename)
  rm(outdir,filename,gsea)
}

```

```{r}
#KEGG
files <- dir("UMAP/NUP98_Only", pattern = "KEGG", recursive = T, full.names = T)
results_KEGG <- lapply(files, read.csv, row.names=1)
names(results_KEGG) <- paste0("Cluster_",1:5)

lapply(results_KEGG, dim)
```

```{r}
unique_pathfindR <- list()
for (i in 1:5){
  gs <- results_KEGG[[i]]$ID
  others <- lapply(c(1:5)[-i],function(x) results_KEGG[[x]]$ID) %>% 
    unlist()
  
  uniq <- gs[! gs %in% others]  
  unique_pathfindR[[paste0("Cluster_",i)]] <- uniq
  rm(uniq, others)
    
}

# unique_pathfindR


# names(results_KEGG)
# names(unique_pathfindR)

# lapply(1:5, function(x) filter(results_KEGG[[x]], ID %in%  unique_pathfindR[[x]]))
```


```{r}
filter(results_KEGG$Cluster_1,!ID %in% results_KEGG$Cluster_2$ID)  %>% 
  slice(1:5) #%>%
  # pull(Up_regulated)

```

```{r}
filter(results_KEGG$Cluster_2, !ID %in% results_KEGG$Cluster_1$ID) %>%
  slice(1:5)

```

```{r}
filter(results_KEGG$Cluster_4,!ID %in% results_KEGG$Cluster_5$ID)  %>% 
  slice(1:5) #%>%
  # pull(Up_regulated)

```

```{r}
filter(results_KEGG$Cluster_5, !ID %in% results_KEGG$Cluster_4$ID) %>%
  slice(1:5)

```

```{r}
forEnrplots <- lapply(names(results_KEGG), function(x) mutate(results_KEGG[[x]], 
                                                              Cluster=as.numeric(str_split_fixed(x, "_",n=2)[,2]),
                                                              Down_regulated="")) %>% 
  bind_rows() %>% 
  filter(!grepl("infection|virus|Hepatitis|Toxoplasmosis|Shigellosis|measles|cardiomyopathy|invasion", 
                                           Term_Description,ignore.case = T)) %>% 
  as.data.frame()

head(forEnrplots)
# dim(forEnrplots) #252   9
```

```{r}
uniq_paths_Enrplots <- lapply(names(results_KEGG),
                              function(x) filter(results_KEGG[[x]], ID %in%  unique_pathfindR[[x]]) %>% 
                                            filter(!grepl("infection|virus|Hepatitis|Toxoplasmosis|Shigellosis|measles|cardiomyopathy|invasion|Cardiac muscle|Amoebiasis|cardiomyocytes", 
                                                          Term_Description,ignore.case = T)) %>% 
                                            slice(1:5) %>%
                                            mutate(Cluster=as.numeric(str_split_fixed(x, "_",n=2)[,2]),
                                                   Down_regulated="")) %>% 
  bind_rows() %>% 
  select(ID,Cluster, everything()) %>%
  as.data.frame()

uniq_paths_Enrplots
# head(uniq_paths_Enrplots)
# dim(forEnrplots) #252   9


# write.csv(uniq_paths_Enrplots,"UMAP/NUP98_Only/UMAP_NUP98.R_Clusters_Unique_KEGG_pathways.csv",
#           row.names = FALSE)

```


```{r fig.height=7, fig.width=8}
source(file.path(SCRIPTS,"example_scripts/dot_plot_GSEA_PathFindR.R"))

g <- enrichment_chart(result_df = uniq_paths_Enrplots,
                 plot_by_cluster = TRUE,
                 top_terms = 5) 


g <- g +
  labs(title="Significantly Enriched Pathways") +
  theme(text=element_text(size=12),
      axis.text.y = element_text(size=14, color = "black"),
      axis.text.x = element_text(size=12, color="black"),
      legend.background=element_blank(),
      legend.direction = 'vertical',
      legend.position = 'top',
      legend.justification='left')

g


```

```{r}
ggsave(filename = "Figures/NUP98.R_Cluster_Unique_KEGG_pathways.pdf",
      plot = g, device = "pdf",
      height = 8, width = 8)
```

```{r}
input <- samps_nup_majorFusions %>% 
  as.data.frame() %>% 
  mutate_at(vars(SNVs), ~ifelse(is.na(.), Group, .)) %>% 
  filter(AML_Subtype != "ETS-Fusion") %>%   #too heterogenous
  mutate(USI1=USI,
         USI=Sample) %>% 
  mutate(AML_Subtype=factor(AML_Subtype, levels=c(
                            "NUP98-X",
                           "NUP98-KDM5A",
                           "NUP98-NSD1",
                           "CBFA2T3-GLIS2",
                           "CBFB-MYH11",
                           "DEK-NUP214",
                           "KMT2A",
                           "RBM15-MKL1",
                           "RUNX1-RUNX1T1",
                           "CD34_PB", 
                           "NBM"))) %>% 
  arrange(AML_Subtype) %>% 
  set_rownames(.$Sample) 


cts_all <- cts[, input$Sample]
cts_all <- as.matrix(cts_all[rowSums(cts_all >= 10) >= 10, ])

dim(cts_all) #29854   988

input <- filter(input, !is.na(AML_Subtype))

# lapply(Cols, function(x) table(input[,x],
#                                useNA='ifany'))

# dim(input)
# table(input$Group)
table(input$AML_Subtype)
```

## Clustering

```{r}
library(edgeR)
dge <- DGEList(counts=cts_all)
dge <- calcNormFactors(dge,method = "TMMwsp")

CPM <- edgeR::cpm(dge,log=FALSE,normalized.lib.sizes=TRUE, prior.count=1)
logCPM <- edgeR::cpm(dge,log=TRUE,normalized.lib.sizes=TRUE, prior.count=1)


dim(logCPM) #29854   988
# head(logCPM[,1:5])

# Mean vs Dispersion Feature Selection 
obj <- seqGlue::calc_dispersion(cts_all, removeOutliers = TRUE) #removes outlier genes/transcripts based on cooks distance
gc()

sg_all <- seqGlue::get_selected_genes(seqGlue::select_genes(obj, top_n=NULL))
```

```{r}
cc_hmap<- list(
          "SNVs"=c("CD34_PB"="grey", "NBM"="black",
                    "OtherAML"="aliceblue",
                    "FLT3-ITD"="#A6CEE3", 
                    "FLT3-ITD/WT1"="#1F78B4", 
                    "NPM1"="#B2DF8A","WT1"="#33A02C"),
           
           "AML_Subtype"=c("CD34_PB"="grey", "NBM"="black",
                           "CBFA2T3-GLIS2"="blue",
                           "CBFB-MYH11"="red",
                           "DEK-NUP214"="peru",
                           "KMT2A"="khaki2",
                           "NUP98-X"="green1",
                           "NUP98-KDM5A"="magenta",
                           "NUP98-NSD1"="steelblue1",
                           "RBM15-MKL1"="purple3",
                           "RUNX1-RUNX1T1"="sienna"), 
          
          "Any_chr13_Abnormality"=c("CD34_PB"="grey", 
                               "NBM"="black",
                               "chr13_Abnormality"="darkslategrey",
                               "None"="aliceblue",
                                "Unknown"="cornsilk3"))
```

```{r}
dends_res <- make_dends(TMMCPM=logCPM[sg_all, input$Sample])


names(dends_res)
```

```{r fig.height=6, fig.width=20}
#The heatmap dendrogram rotation changed since I last ran it - but results are the same. 
mat_temp <- matrix(ncol=ncol(dends_res$TMMCPM), nrow=1, dimnames = list("", labels(as.dendrogram(dends_res$samp.c1)))) %>%
  as.data.frame() %>% 
  select(TARGET.20.PAWMIX.09A.01R:TARGET.20.PASFLG.09A.01R,
         TARGET.00.RO02590.14A.01R:TARGET.00.RO02748.14A.01R,
         TARGET.00.BM5751.14A.01R:TARGET.00.BM4473.14A.01R,
         TARGET.00.R003616.34POS.10A.01R:TARGET.00.R003624.34POS.14A.01R, 
         TARGET.20.PAWNMU.09A.01R:TARGET.20.PASRLS.09A.01R, 
         TARGET.20.PAVUDH.09A.01R:TARGET.20.PAVSNL.09A.01R, 
         TARGET.20.PASWAJ.09A.01R:TARGET.20.PARZYL.03A.01R, 
         TARGET.20.PAWAFT.03A.01R:TARGET.20.PAXGGM.03A.01R)

# pdf("original_order_dend.pdf",height = 6, width = 25)
par(mar=c(25,4,10,0), cex=0.2,cex.lab=0.5)
d_orig <- dendextend::rotate(as.dendrogram(dends_res$samp.c1),  order=c(ncol(dends_res$TMMCPM):1))
input_orig <- input %>% 
  select(Sample, AML_Subtype) %>% 
  left_join(., data.frame(AML_Subtype=names(cc_hmap$AML_Subtype),
                          Color=cc_hmap$AML_Subtype), 
            by="AML_Subtype") %>% 
  mutate(Sample=factor(Sample, levels=labels(d_orig)),
         AML_Subtype=as.factor(AML_Subtype)) %>% 
  arrange(Sample)
d_orig <- dendextend::color_branches(dend=d_orig, clusters = as.numeric(input_orig$AML_Subtype), col=unique(input_orig$Color))
plot(d_orig)
# dev.off()



#Re-order the clusters 
d_ordered <- dendextend::rotate(as.dendrogram(dends_res$samp.c1),  order=colnames(mat_temp))
input_ordered <- input_orig %>% 
  mutate(Sample=factor(Sample, levels=labels(d_ordered))) %>% 
  arrange(Sample)
d_ordered <- dendextend::color_branches(dend = d_ordered, clusters = as.numeric(input_ordered$AML_Subtype), col=unique(input_ordered$Color))
plot(d_ordered)
```




### Heatmap 

```{r}
# "NUP98-X"="green1","NUP98-KDM5A"="magenta","NUP98-NSD1"="steelblue1"
cols.colorbar <- c("SNVs","Any_chr13_Abnormality","AML_Subtype")
                   # "NUP98.Rearranged.Groups",
                   # "deletion_chr13_groups","monosomy_trisomy_13_groups",
                   # "Translocation_13_groups","M7_AML")


col.annots <- create_HA_Labs_Hmap(expn=logCPM[,input$Sample],
                                  geneList = sg_all,
                                  cc=cc_hmap,
                                  CDE = input,
                                  cols = cols.colorbar)
# col.annots$annoColumn
```

```{r}
# mat <- log2(sf.scaled_all[sg_all,input$Sample]+1)
mat <- logCPM[sg_all, input$Sample]
hmap <- ComplexHmap(mat=mat,
                    dge_dendrograms.res = dends_res,
                    scale = TRUE,
                    name = "AML",
                    hmap_anno_obj = col.annots$annoColumn,
                    samp_dend_order = colnames(mat_temp))


dim(mat)
```

```{r fig.height=10, fig.width=25}
# svg("TARGET_AML_NUP98-R_Heatmap.svg",width = 16, height = 8) #SVG SUCKS it makes a goddamn png NOT an SVG
# pdf("Figures/Heatmaps/TARGET_AML_NUP98-R_Heatmap_v3.pdf",width = 24, height = 8)
draw(hmap)
# dev.off()
```



## Recolor for del13q 

```{r}
umap_all_aml <- readRDS("UMAP/TARGET_AML_sg6818_UMAP_Results.RDS")

# dim(umap_all_aml$umap_res)
# table(umap_all_aml$umap_res$AML_Subtype)
# table(umap_all_aml$umap_res$Group)
# length(umap_all_aml$input_features)
```

```{r fig.height=5, fig.width=10}
colors_del13 <- c("NUP98-KDM5A" = "magenta",
                  "NUP98-KDM5A/chr13_abn" = "darkorchid4",
                 "NUP98-NSD1" = "steelblue1",
                 "NUP98-NSD1/chr13_abn"="steelblue4",
                 "NUP98-X" = "green1",
                 "NUP98-X/chr13_abn"="darkgreen",
                 "OtherAML" = "grey80",
                 "OtherAML/chr13_abn"="cyan3")


# colors_del13 <- c("NUP98-KDM5A" = "magenta",
#                   "NUP98-KDM5A_del13q" = "darkorchid4",
#                  "NUP98-NSD1" = "steelblue1",
#                  "NUP98-X" = "green1",
#                  "OtherAML" = "grey80",
#                  "OtherAML_del13q" = "darkslategray3")
#                  # "NBM" = "gray50")


par(mar=c(10,10,10,10))
barplot(rep(1, length(colors_del13)),col=colors_del13, las=2, names.arg = names(colors_del13))
```


```{r}
umap_res_recolor <- umap_all_aml$umap_res %>% 
  select(Sample, x:z) %>% 
  inner_join(., sample_info, by="Sample") %>%
  filter(!USI %in% inelig) %>% 
  
  mutate(chr13_Abn_groups=case_when(
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "NUP98-X" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "NUP98-KDM5A" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    Any_chr13_Abnormality == "chr13_Abnormality" & NUP98.Rearranged.Groups == "OtherAML" ~ paste(NUP98.Rearranged.Groups, "chr13_abn", sep="/"),
    TRUE ~ NUP98.Rearranged.Groups)) %>% 
  
  # mutate(Chr13_Abn_Groups=case_when(
  #   Any_chr13_Abnormality == "chr13_Abnormality" ~ paste(NUP98.Rearranged.Groups, "Abn_Chr13", sep="/"),
  #   TRUE ~ NUP98.Rearranged.Groups)) %>% 
  # mutate_at(vars("Any_chr13_Abnormality"), ~gsub("OtherAML","None", .)) %>% 
  # mutate_at(vars(NUP98.Rearranged.Groups), ~case_when(
  #   Any_chr13_Deletion == "chr13_deletion" & . == "NUP98-KDM5A" ~ paste(., "del13q", sep="_"),
  #   Any_chr13_Deletion == "chr13_deletion" & . == "OtherAML" ~ paste(., "del13q", sep="_"),
  #   TRUE ~ .)) %>% 
  # mutate_at(vars(NUP98.Rearranged.Groups), ~factor(., levels=rev(names(colors_del13)))) %>% 
  set_rownames(.$Sample)

table(umap_res_recolor$chr13_Abn_groups)
umap_nups_others <- umap_all_aml
umap_nups_others$umap_res <- umap_res_recolor


# table(umap_res_recolor$Time_point)
table(umap_res_recolor$chr13_Abn_groups)
```

#### plot.ly

```{r message=FALSE}
library(plotly)

scatter3d <- scatter_plots_3d(umap_workflow_res = umap_nups_others, 
                              Group_Column = "chr13_Abn_Groups",
                              Cols = c("Sample",
                                       "Primary.Fusion",
                                       "Primary.CNV",
                                       "Any_chr13_Abnormality",
                                       "deletion_chr13",
                                       "translocation_13",
                                       "monosomy_trisomy_13",
                                       "Age.in.years",
                                       "Overlap.Mutation.Info",
                                       "EFS.event.type.ID"),
                              cc = colors_del13,
                              blackbg = FALSE,
                              ptsize = 3)
```

```{r fig.height=10, fig.width=10}
# scatter3d

# bgcol <- rgb(0,0,0)
# bgcol <- rgb(1,1,1)
# htmlwidgets::saveWidget(as_widget(scatter3d),
#                         "TARGET_AML_NUP98_Del13q_3Dscatter_v2.html",
#                         selfcontained = TRUE,
#                         background = bgcol,
#                         knitrOptions = list(dpi=1200, 
#                                             fig.height = 15,
#                                             fig.width=15,
#                                             out.width=15, 
#                                             out.height=15))
```

#### Scatter3D


```{r fig.height=7, fig.width=7}
colors_3d_other <- umap_nups_others$umap_res  %>% 
  filter(!grepl("NUP98", chr13_Abn_groups)) %>% 
  left_join(., data.frame(color=colors_del13) %>% 
              rownames_to_column("chr13_Abn_groups"),
            by="chr13_Abn_groups") %>% 
  mutate_at(vars(color), ~sapply(., make_alpha, percent=90)) 

colors_3d_NUP98 <- umap_nups_others$umap_res  %>% 
  filter(grepl("NUP98|chr13_abn", chr13_Abn_groups)) %>% 
  left_join(., data.frame(color=colors_del13) %>% 
              rownames_to_column("chr13_Abn_groups"),
            by="chr13_Abn_groups") %>% 
   mutate_at(vars(color), ~sapply(., make_alpha, percent=40)) %>% 
  select(Sample, NUP98.Rearranged.Groups, chr13_Abn_groups, color,x,y,z) %>% 
  arrange(chr13_Abn_groups)


t <- -200
p <- 20

# theta=-45
# theta=100

# pdf("Figures/UMAP_PCA/TARGET_AML_NUP98.Rearranged_with_OtherAML_plot3D_scatter_3D_theta-200_phi15_UMAP.pdf", height = 10, width = 10)
plot3D::scatter3D(x=colors_3d_other$x, y=colors_3d_other$y, z=colors_3d_other$z,
                   bty = "b2",cex=1.5,pch=19, bg="black",
                  colvar=as.numeric(as.factor(colors_3d_other$chr13_Abn_groups)),
                  col=unique(colors_3d_other$color),
                  colkey=FALSE,
                  theta=t,phi=p, 
                  scale = FALSE)

plot3D::scatter3D(x=colors_3d_NUP98$x, y=colors_3d_NUP98$y, z=colors_3d_NUP98$z,
                   bty = "b2",cex=1.5, 
                  pch=19, bg="black",
                  colvar=as.numeric(as.factor(colors_3d_NUP98$chr13_Abn_groups)),
                  col=unique(colors_3d_NUP98$color),
                  colkey=FALSE,
                  theta=t,phi=p, 
                  scale = TRUE,
                  add = TRUE)
# dev.off()
```



#### ggplot 

```{r fig.height=6, fig.width=5}
umap_subset <- filter(umap_nups_others$umap_res, 
                             !grepl("OtherAML$|KDM5A", chr13_Abn_groups))

umap_kdm5A_subset <- filter(umap_nups_others$umap_res, 
                             grepl("KDM5A", chr13_Abn_groups)) %>% 
  droplevels()


NUP98_del13_dim12 <- ggplot(filter(umap_nups_others$umap_res, grepl("OtherAML$", chr13_Abn_groups)),
                      aes(x=x,y=y,color=chr13_Abn_groups)) +
  
      geom_point(size=0.75,alpha=0.6,shape=19) +
     geom_point(data=umap_subset,
                 size=2.0,alpha=0.5,shape=19)+
      geom_point(data=umap_kdm5A_subset,
                 size=2.0,alpha=0.5,shape=19)+
      scale_color_manual(values=colors_del13) +
      scale_fill_manual(values=colors_del13) +
      # scale_y_continuous(breaks = seq(-5,5,by=5)) +
      # scale_x_continuous(breaks = seq(-3,8,by=3), limits = c(-3,8)) +
      labs(x="UMAP 1", y="UMAP 2", 
       color="NUP98 Fusion Status",
       fill="NUP98 Fusion Status"
       # title="Pediatric AML with Chr13 Abnormalities"
       ) +
    guides(color = guide_legend(override.aes = list(size=6), ncol=2, 
                                title.position="top", title.hjust = 0.5),
         fill = guide_legend(override.aes = list(size=6), ncol=2, 
                             title.position="top", title.hjust = 0.5)) +

    theme_classic() +
    theme(axis.text = element_text(size=16, color="black"),
        axis.title = element_text(size=18), 
        legend.text = element_text(size=14), 
        legend.title = element_text(size=14),
        legend.position = "bottom")
      

# pdf("Figures/Chr13_deletions/TARGET_AML_NUP98_chr13_dim1_UMAP_2Dscatter.pdf", height = 6, width=5)
NUP98_del13_dim12
# dev.off()

# saveRDS(NUP98_dim12,"TARGET_AML__UMAP_dim1_2Dscatter.RDS")
```

```{r fig.height=5, fig.width=5}
umap_kdm5A_subset <- filter(umap_nups_others$umap_res, 
                             grepl("KDM5A", NUP98.Rearranged.Groups)) %>% 
  droplevels()

#Use DBScan for density based clustering! 
#http://www.sthda.com/english/wiki/wiki.php?id_contents=7940 
# cl <- (kmeans(umap_kdm5A_subset[,c("x","y")],2)) 
# cl.df <- data.frame(cluster=factor(cl$cluster),
#                     Center=cl$centers)

kdm5A_del13_dim12 <- ggplot(umap_kdm5A_subset,
                      aes(x=x,y=y,color=NUP98.Rearranged.Groups)) +
  
      geom_point(size=2.5,alpha=0.75) +
      # geom_point(data=centers, aes(x=V1,y=V2, color="Center")) +
      geom_point(data=cl.df,
                 aes(x=Center.x,y=Center.y, fill=cluster),
                 shape=21, size=52, alpha=.3, legend=FALSE, 
                 inherit.aes = F) +
      scale_color_manual(values=colors_del13) +
      scale_fill_manual(values=colors_del13) +
      # scale_y_continuous(breaks = seq(-5,5,by=5), limits = c(-5,5)) +
      # scale_x_continuous(breaks = seq(-3,8,by=3), limits = c(-3,8)) +
      xlab("UMAP_1") + ylab("UMAP_2") +
      labs(title="") +
      theme_classic() +
      guides(fill = 'none', color='none') +
      theme(plot.margin = margin(r = 5, unit="mm"))
      

# pdf("TARGET_AML__UMAP_dim1_2Dscatter.pdf", height = 7, width=10)
kdm5A_del13_dim12
```

```{r fig.height=5, fig.width=10}
# NUP98_del13_dim12 + kdm5A_del13_dim12
```


### 2D Scatter plots


```{r}
umap_allDX <- readRDS(file.path(PROJHOME,"UMAP/NUP98_Only/TARGET_AML_sg8780_NUP98.Rearr_log2Counts_UMAP_NUP98_Results_7.16.20.RDS"))

dim(umap_allDX$umap_res)
table(umap_allDX$umap_res$AML_Subtype)
table(umap_allDX$umap_res$Group)
length(umap_allDX$input_features)

# umap_allDX$umap_res$cluster_k5
```

```{r fig.height=3}
flevels <- rev(c("NUP98-X", "NUP98-NSD1","NUP98-KDM5A"))

res <- umap_allDX$umap_res %>%
  mutate_at(vars(NUP98.Rearranged),
                 ~factor(., levels=flevels))


NUP98_colors <- cc$NUP98.Rearranged[flevels]
par(mar=c(8,4,4,2))
barplot(rep(1,length(NUP98_colors)), col=NUP98_colors, 
        names.arg = names(NUP98_colors), las=2)
```

```{r}
labs <- c("Single Mutant"="NUP98 single mutant",
          "Double Mutant"="NUP98 double mutant", 
          "KMT2A"="KMT2A-rearranged", "RUNX1-RUNX1T1"="RUNX1-RUNX1T1",
          "CBFB-MYH11"="CBFB-MYH11", 
          "NUP98-NSD1"="NUP98-NSD1",
          "DEK-NUP214"="DEK-NUP214", 
          "NUP98-KDM5A"="NUP98-KDM5A",
          "CBFA2T3-GLIS2"="CBFA2T3-GLIS2", 
          "NPM1"="NPM1",
          "FLT3-ITD"="FLT3-ITD",
          "No.Primary.Fusion.CNV"="No Fusion/CNV",
          "AML"="Other AML")

thm <- theme(
  # panel.background = element_rect(color = "black", fill="black"),
            axis.text = element_text(size=22, color="black"),
            axis.title = element_text(size=22, color="black")) 

ellipse_dat <- res %>% 
  filter(grepl("RUNX1|DEK|NUP98|Mutant|KMT2A|CBFB|GLIS2|NPM1|FLT3",
               NUP98.Rearranged)) %>% 
  mutate(ellipse_groups=ifelse(grepl("Mutant",as.character(NUP98.Rearranged)),
                               "NUP98", as.character(NUP98.Rearranged))) %>% 
  droplevels()



ellipse_colors <- set_names(c("red","dodgerblue","forestgreen","darkorchid3", "gold"),
                            levels(res$cluster_k5))
# ellipse_colors <- NUP98_colors[-c(1:2,10:11)]
# ellipse_colors["NUP98"] <- "dodgerblue2"
```


```{r fig.height=7, fig.width=10}
NUP98_dim12 <- ggplot(res,
                      aes(x=x,y=y,color=AML_Subtype, group=cluster_k5)) +
      stat_ellipse(data=res,
                   aes(x=x,y=y, fill=cluster_k5, group=cluster_k5),
                   geom = "polygon",
                   size=1.5,
                   alpha=0.15,
                   level = 0.98,
                   inherit.aes = FALSE) +
      # geom_point(size=3.0,alpha=1,shape=21, stroke = 0.2)+
      geom_point(size=3.0,alpha=1,shape=19, stroke=0.2)+
      scale_color_manual(values=NUP98_colors, labels=labs) +
      scale_fill_manual(values=ellipse_colors) +
      scale_y_continuous(breaks = seq(-5,5,by=5)) +
      xlab("UMAP_1") + ylab("UMAP_2") +
      labs(title="") +
      theme_classic() +
      thm +
      guides(fill = 'none', color='none') +
      # guides(color=guide_legend(override.aes = list(size=7)), fill='none') +
      theme(plot.margin = margin(r = 5, unit="mm"))
      

NUP98_dim12
```

```{r fig.height=7, fig.width=20}
NUP98_dim13 <- ggplot(res,shape=21,
                      aes(x=x,y=z,color=AML_Subtype,group=cluster_k5)) +
   stat_ellipse(data=res,
                   aes(x=x,y=z, fill=cluster_k5, group=cluster_k5),
                   geom = "polygon",
                   size=1.5,
                   alpha=0.15,
                   level = 0.95,
                   inherit.aes = FALSE) +
      # geom_point(size=3.0,alpha=1,shape=21, stroke = 0.2)+
      geom_point(size=3.0,alpha=1,shape=19, stroke=0.2)+
      scale_color_manual(values=NUP98_colors, labels=labs) +
      scale_fill_manual(values=ellipse_colors) +

      xlab("UMAP_1") + ylab("UMAP_3") +
      scale_y_continuous(breaks = seq(-5,5,by=5)) +
      labs(title="") +
      theme_classic() +
      thm +
      theme(legend.text = element_text(size=22),
            legend.title = element_blank()) 
      # guides(color = guide_legend(override.aes = list(size=7)), fill='none') 

# NUP98_dim13

# pdf(file.path(PROJHOME,"TARGET_AML_NUP98.R_UMAP_dim123.pdf"), height = 7, width = 20)
grid.arrange(grobs=list(NUP98_dim12,NUP98_dim13),
             ncol=2, widths=c(1,1.25))
# dev.off()
```


### 3D Plot.ly 

```{r message=FALSE}
library(plotly)
```

```{r}
info <- paste(res$Sample, 
              paste0("Mutation Category: ",res$Mutations.Category),
              sep="\n") %>% 
  ifelse(!is.na(res$Primary.Fusion.CNV), 
         paste(., paste0("Primary Fusion CNV: ",res$Primary.Fusion.CNV), sep="\n"), .) %>% 
  ifelse(!is.na(res$Overlap.Mutation.Info),
         paste(., gsub("/", "\n", res$Overlap.Mutation.Info), sep="\n"), .) %>%
  gsub("OtherAML","", .) %>%
  ifelse(!is.na(res$Age.in.years),
         paste( ., paste("Age:", round(res$Age.in.years,digits = 1),"yrs"), sep="\n"), .) %>%
  ifelse(!is.na(res$EFS.event.type.ID),
         paste(., res$EFS.event.type.ID, sep="\n"), .)  %>%
  gsub("Unknown","",.) %>%
  set_names(., res$Sample)
length(info)
```

```{r fig.width=16}
textcol <- rgb(0,0,0)
bgcol <- rgb(1,1,1)

p.NUP98 <- plot_ly() %>% 
  #plot diagnostic and normal samples
  add_trace(data=res,
            x = ~x, y = ~y, z = ~z,
            color = ~NUP98.Rearranged,
            colors = NUP98_colors,
            type='scatter3d',
            mode='markers',
            text=info,
            hoverinfo='text',
            showlegend=TRUE,
            marker=list(size=4.5,
                        opacity=0.5),
                        # line=list(color=col2hex("grey80"),
                        #           width=0.25)),
            # line=list(size=1, color=col2rgb("grey80")),
            inherit = TRUE) %>%
  # title=list(text="Pediatric AML Clustering By Gene Expression",
  layout(font = list(color=textcol,
                                size=10),
         scene = list(xaxis = list(title = 'UMAP_1',
                                   color=textcol,
                                   size=10,
                                   backgroundcolor=bgcol,
                                   showbackground=TRUE,
                                   showgrid=TRUE,
                                   gridcolor=textcol,
                                   tickcolor=textcol),
                     yaxis = list(title = 'UMAP_2',
                                  color=textcol,
                                  size=18,
                                  backgroundcolor=bgcol,
                                  showbackground=TRUE,
                                  showgrid=TRUE,
                                  gridcolor=textcol,
                                  tickcolor=textcol),
                     zaxis = list(title = 'UMAP_3',
                                  color=textcol,
                                  size=10,
                                  backgroundcolor=bgcol,
                                  showbackground=TRUE,
                                  showgrid=TRUE,
                                  gridcolor=textcol,
                                  tickcolor=textcol),
                     bgcolor=bgcol,
         legend=list(font=list(size=12, color=textcol),
                     itemsizing="constant"),
         plot_bgcolor=bgcol,
         paper_bgcolor=bgcol))
 
# p.NUP98
```


#### ggplot


```{r}
# devtools::install_github("AckerDWM/gg3D")
library(gg3D)
library(cowplot)
# library(svglite)
```

```{r fig.height=7, fig.width=10}
# NOTE: Cannot do the standard 'hacky' way of subsetting your ggplot data and adding the group of interest a second layer, such as to make the points bigger or add stroke etc. 
# 
# Instead, use the whole dataset as input for a second layer for a group of interest. Change all unwanted color/fill values to NA. Then set the scale_XXX_disrete/scale_XXXX_manual with na.value=NA and/or with na.translate=FALSE to be transparent. Othersize the axes/rotation do not match up because a new min/max for x,y,z coors are used to draw the axes. 


theta=-120
phi=-10
include_others=TRUE
#good anlges
#t=90,phi=10
#t=-110,phi=-20


bgCol <- "white"
txtCol <- "black"

p <- ggplot(data=res, 
            mapping=aes(x=x, y=y,z=z, 
                        fill=NUP98.Rearranged)) +
    axes_3D(theta=theta, phi=phi, color=txtCol)

rotated <-
  p + stat_3D(data=other_aml,
          aes(x=x, y=y, z=z), color="grey80", #fill=aml
          geom = "point", theta=theta,phi=phi,
          shape=19,size=3.0, alpha=0.3,stroke=0, inherit.aes = FALSE)  

rotated <- rotated + stat_3D(geom = "point", theta=theta,phi=phi,
          shape=21,size=3.0,
          color="grey50",
          stroke=0.2,alpha=0.75) +
  scale_color_manual(values=NUP98_colors, labels=labs,na.value=NA,
                     na.translate = FALSE)  +
  scale_fill_manual(values=NUP98_colors, labels=labs,na.value=NA,
                     na.translate = FALSE)  +
  guides(fill=guide_legend(override.aes = list(size=7,
                                               alpha=1)),
         color='none') +
  labs_3D(labs=c("UMAP_1", "UMAP_2", "UMAP_3"),
    hjust=c(-0.25,-2,-2), vjust=c(10,3.5,31.0),
    angle=c(0,90,0), size=6, color=txtCol) +
  labs(title="UMAP Clustering of Pediatric AML \n by Gene Expression") +
  theme_void() +
  theme(panel.background = element_rect(color=bgCol, fill=bgCol),
        plot.title = element_text(hjust=0.5, vjust=0.5, size = 20),
        legend.text = element_text(size=20),
        legend.title = element_blank())

rotated
```



# Expression of gene partners 

```{r}
gene_partners <- read.csv(file.path(PROJHOME,"TARGET_AML_NUP98_gene_pairs.csv"))

gene_partners
```

```{r}
TPMs.sub <- TPMs[,c("Gene",samps_nup_majorFusions$Sample)] %>% 
  filter(Gene %in% c(gene_partners$geneSymbol, "MLL")) %>% 
  gather(Sample,TPM, -Gene) %>% 
  mutate(log2TPM=log2(TPM+1)) %>% 
  left_join(., samps_nup_majorFusions, by="Sample")


dim(TPMs.sub)
head(TPMs.sub)
# TPMs.sub[,1:5]

unique(TPMs.sub$Gene)
# c(gene_partners$geneSymbol,"KMT2A")
```

```{r fig.height=10, fig.width=15}
ggplot(TPMs.sub, aes(x=NUP98.Rearranged.Groups, y=log2TPM, fill=NUP98.Rearranged.Groups)) +
  geom_boxplot() +
  facet_wrap(~Gene, scales = 'free_y') +
  theme(axis.text.x = element_text(angle=25, vjust=1, hjust=1))
```


# Session Information

```{r}
sessionInfo()
```

